File: app/src/main/java/com/moh/mhealth/pages/Diagnostics4.java
package com.moh.mhealth.pages;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.moh.mhealth.Global_Helper;
import com.moh.mhealth.Header;
import com.moh.mhealth.Patient;
import com.moh.mhealth.R;

public class Diagnostics4 extends AppCompatActivity implements Header {

    EditText diagField, treatField;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_diagnostics4);

        diagField = (EditText) findViewById(R.id.diag4_diag);
        treatField = (EditText) findViewById(R.id.diag4_diag);

        loadPatientData();

        Button nextBtn = (Button) findViewById(R.id.diag4_finish);

        nextBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                updatePatientData();
                savePatient();

                startActivity(new Intent(getApplicationContext(), Global_Helper.endPatient()));
            }
        });
    }

    private void updatePatientData() {
        Global_Helper.addDiag4(diagField.getText().toString(),
                               treatField.getText().toString());
    }

    private void savePatient() {

    }

    private void loadPatientData() {
        Patient patient = Global_Helper.getCurrentPatient();

        TextView patient_name = (TextView) findViewById(R.id.header_name);
        patient_name.setText(patient.getName());

        diagField.setText(patient.getDiag());
        treatField.setText(patient.getTreat());
    }


    @Override
    public void cancel(View view) {
        startActivity(new Intent(getApplicationContext(), Global_Helper.endPatient()));
    }

    @Override
    public void moveBack(View view) {
        startActivity(new Intent(getApplicationContext(), Global_Helper.prevDiag()));
    }
}

File: app/src/main/java/com/moh/mhealth/pages/Diagnostics4.kt
package com.moh.mhealth.pages

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.activity.viewModels
import com.moh.mhealth.*
import com.moh.mhealth.patientdatabase.PatientViewModel

class Diagnostics4 : AppCompatActivity(), Header {
    private var diagField: EditText? = null
    private var treatField: EditText? = null

    private val patientViewModel: PatientViewModel by viewModels { PatientViewModel.Factory }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_diagnostics4)

        diagField = findViewById<View>(R.id.diag4_diag) as EditText
        treatField = findViewById<View>(R.id.diag4_treat) as EditText

        loadPatientData()

        val nextBtn = findViewById<View>(R.id.diag4_finish) as Button
        nextBtn.setOnClickListener {
            updatePatientData()
            savePatient()
            startActivity(Intent(applicationContext, GlobalHelper.endPatient()))
        }
    }

    private fun updatePatientData() {
        GlobalHelper.addDiag4(
            diagField!!.text.toString(),
            treatField!!.text.toString()
        )
    }

    private fun savePatient() {
        patientViewModel.insert(GlobalHelper.currentPatient!!)
        Toast.makeText(this, "Patient Saved.", Toast.LENGTH_SHORT).show()
    }

    private fun loadPatientData() {
        val patient = GlobalHelper.currentPatient!!
        val patientName = findViewById<View>(R.id.header_name) as TextView
        patientName.text = patient.name
        diagField!!.setText(patient.diag)
        treatField!!.setText(patient.treat)
    }

    override fun cancel(view: View) {
        startActivity(Intent(applicationContext, GlobalHelper.endPatient()))
    }

    override fun moveBack(view: View) {
        startActivity(Intent(applicationContext, GlobalHelper.prevDiag()))
    }
}